.github/CHANGELOG.md

```diff
@@ -2,6 +2,8 @@
 
 <h3>New features since last release</h3>
 
+* [Placeholder for measurement grouping features]
```

Comment:
Please see my suggested description:

<h3>New features since last release</h3>

* Functionality for grouping simultaneously measurable Pauli word observables included in newly introduced `grouping` module. [(#761)](https://github.com/PennyLaneAI/pennylane/pull/761) All utility functions required for measurement reduction by qubit-wise-commuting (QWC) grouping are included. See [arXiv:1907.03358](https://arxiv.org/abs/1907.03358) for a detailed description of the QWC grouping technique.

* The `optimize_measurements` function will take as input a list of Pauli word observables and their corresponding coefficients (if any), and will return the partitioned Pauli terms diagonalized in the measurement basis and the corresponding diagonalizing circuits. 

 Below is an example of QWC measurement optimization applied to a H<sub>2</sub> molecular electronic Hamiltonian in STO-3G basis:

 ```python
 import pennylane as qml

 h, nr_qubits = qml.qchem.generate_hamiltonian(
    mol_name='h2',
    mol_geo_file='h2.xyz',
    mol_charge=0,
    multiplicity=1,
    basis_set='sto-3g',
    mapping='jordan_wigner')
 ```

 for which the Hamiltonian `h` consists of 15 
 Pauli words,

 ```
>>> h.ops 
[Tensor(Identity(wires=[0])), 
Tensor(PauliZ(wires=[0])), Tensor(PauliZ(wires=[1])), Tensor(PauliZ(wires=[2])), Tensor(PauliZ(wires=[3])), Tensor(PauliZ(wires=[0]), PauliZ(wires=[1])), Tensor(PauliY(wires=[0]), PauliX(wires=[1]), PauliX(wires=[2]), PauliY(wires=[3])), Tensor(PauliY(wires=[0]), PauliY(wires=[1]), PauliX(wires=[2]), PauliX(wires=[3])), Tensor(PauliX(wires=[0]), PauliX(wires=[1]), PauliY(wires=[2]), PauliY(wires=[3])), Tensor(PauliX(wires=[0]), PauliY(wires=[1]), PauliY(wires=[2]), PauliX(wires=[3])), Tensor(PauliZ(wires=[0]), PauliZ(wires=[2])), Tensor(PauliZ(wires=[0]), PauliZ(wires=[3])), Tensor(PauliZ(wires=[1]), PauliZ(wires=[2])), Tensor(PauliZ(wires=[1]), PauliZ(wires=[3])), Tensor(PauliZ(wires=[2]), PauliZ(wires=[3]))]
>>> h.coeffs
[-0.32760819  0.13716573  0.13716573 -0.13036292 -0.13036292  0.15660062
  0.04919765 -0.04919765 -0.04919765  0.04919765  0.10622904  0.15542669
  0.15542669  0.10622904  0.16326769]
 ```

 Using the QWC measurement optimization results 
 in a partitioning into 5 groupings:

 ```python
 rotations, grouped_ops, grouped_coeffs = optimize_measurements(h.ops, h.coeffs, grouping='qwc')

 ```

 The diagonalizing circuits of `rotations` 
 correspond to the diagonalized Pauli word groupings 
 of `grouped_ops`,

 ```
>>> rotations
[[RX(1.5707963267948966, wires=[0]), RY(-1.5707963267948966, wires=[1]), RY(-1.5707963267948966, wires=[2]), RX(1.5707963267948966, wires=[3])], [RX(1.5707963267948966, wires=[0]), RX(1.5707963267948966, wires=[1]), RY(-1.5707963267948966, wires=[2]), RY(-1.5707963267948966, wires=[3])], [RY(-1.5707963267948966, wires=[0]), RY(-1.5707963267948966, wires=[1]), RX(1.5707963267948966, wires=[2]), RX(1.5707963267948966, wires=[3])], [RY(-1.5707963267948966, wires=[0]), RX(1.5707963267948966, wires=[1]), RX(1.5707963267948966, wires=[2]), RY(-1.5707963267948966, wires=[3])], []]
>>> grouped_ops
[[Identity(wires=[0]), Tensor(PauliZ(wires=[0]), PauliZ(wires=[1]), PauliZ(wires=[2]), PauliZ(wires=[3]))], [Tensor(PauliZ(wires=[0]), PauliZ(wires=[1]), PauliZ(wires=[2]), PauliZ(wires=[3]))], [Tensor(PauliZ(wires=[0]), PauliZ(wires=[1]), PauliZ(wires=[2]), PauliZ(wires=[3]))], [Tensor(PauliZ(wires=[0]), PauliZ(wires=[1]), PauliZ(wires=[2]), PauliZ(wires=[3]))], [PauliZ(wires=[0]), PauliZ(wires=[1]), PauliZ(wires=[2]), PauliZ(wires=[3]), Tensor(PauliZ(wires=[0]), PauliZ(wires=[1])), Tensor(PauliZ(wires=[0]), PauliZ(wires=[2])), Tensor(PauliZ(wires=[0]), PauliZ(wires=[3])), Tensor(PauliZ(wires=[1]), PauliZ(wires=[2])), Tensor(PauliZ(wires=[1]), PauliZ(wires=[3])), Tensor(PauliZ(wires=[2]), PauliZ(wires=[3]))]]
>>> grouped_coeffs
[[-0.32760818967480876, 0.04919764587136755], [-0.04919764587136755], [-0.04919764587136755], [0.04919764587136755], [0.13716572937099497, 0.13716572937099492, -0.13036292057109117, -0.1303629205710912, 0.15660062488237947, 0.10622904490856078, 0.15542669077992832, 0.15542669077992832, 0.10622904490856078, 0.16326768673564346]]
 ```

 Note that specifying coefficients to `optimize_measurements` is optional. Only `grouping='qwc'` is supported in this addition.

* Pauli word partitioning utilities are performed by the `PauliGroupingStrategy` class in `grouping/group_observables`. An input list of Pauli words can be partitioned into mutually commuting, qubit-wise-commuting, or anticommuting groupings. Partitioning is performed by mapping the Pauli words to graph vertices and the symmetric binary relation (e.g. commutativity) between Pauli words to graph edges, then colouring the complementary graph using heuristic methods. A list of Pauli words and (optionally) their coefficients can be grouped by `group_observables`.

 For instance, given the 16 possible 2-qubit Pauli words, 

 ```python
from pennylane import PauliX, PauliY, PauliZ, Identity
pauli_words = [Identity('a') @ Identity('b'),
               Identity('a') @ PauliX('b'),
               Identity('a') @ PauliY('b'),
               Identity('a') @ PauliZ('b'),
               PauliX('a') @ Identity('b'),
               PauliX('a') @ PauliX('b'),
               PauliX('a') @ PauliY('b'),
               PauliX('a') @ PauliZ('b'),
               PauliY('a') @ Identity('b'),
               PauliY('a') @ PauliX('b'),
               PauliY('a') @ PauliY('b'),
               PauliY('a') @ PauliZ('b'),
               PauliZ('a') @ Identity('b'),
               PauliZ('a') @ PauliX('b'),
               PauliZ('a') @ PauliY('b'),
               PauliZ('a') @ PauliZ('b')]
 ```

 they can be partitioned into anticommutative groupings by the Recursive Largest First (RLF) graph colouring heuristic:

 ```python
from pennylane.grouping.group_observables import group_observables 
groupings = group_observables(pauli_words, grouping_type='anticommuting', method='rlf')
 ```

 resulting in 5 mutually anticommutative groupings:

 ```
>>> groupings
[[Identity(wires=['a'])], [PauliX(wires=['b']), PauliY(wires=['b']), Tensor(PauliX(wires=['a']), PauliZ(wires=['b'])), Tensor(PauliY(wires=['a']), PauliZ(wires=['b'])), Tensor(PauliZ(wires=['a']), PauliZ(wires=['b']))], [PauliZ(wires=['b']), Tensor(PauliZ(wires=['a']), PauliX(wires=['b'])), Tensor(PauliX(wires=['a']), PauliX(wires=['b'])), Tensor(PauliY(wires=['a']), PauliX(wires=['b']))], [PauliX(wires=['a']), PauliY(wires=['a']), PauliZ(wires=['a'])], [Tensor(PauliY(wires=['a']), PauliY(wires=['b'])), Tensor(PauliX(wires=['a']), PauliY(wires=['b'])), Tensor(PauliZ(wires=['a']), PauliY(wires=['b']))]]
 ```

* Various utility functions are included in `grouping/utils` for obtaining and manipulating Pauli words in the binary symplectic vector space representation. 

 For instance, two Pauli words may be converted to their binary vector representation:

 ```python
from pennylane.grouping.utils import pauli_to_binary
from pennylane import PauliX, PauliY, PauliZ
from pennylane.wires import Wires
wire_map = {Wires('a'): 0, Wires('b'): 1}
pauli_vec_1 = pauli_to_binary(PauliX('a') @ PauliY('b'))
pauli_vec_2 = pauli_to_binary(PauliZ('a') @ PauliZ('b'))
 ```

 ```
>>> pauli_vec_1 
[1. 1. 0. 1.]
>>> pauli_vec_2
[0. 0. 1. 1.]
 ```

 Then, as an example, their product up to a phase may be computed by taking the sum of their binary vector representations, and returned to the  operator representation

 ```python
 from pennylane.grouping.utils import binary_to_pauli
product = binary_to_pauli((pauli_vec_1 + pauli_vec_2) % 2, wire_map)
 ```

 giving

 ```
>>> product
Tensor product ['PauliY', 'PauliX']: 0 params, wires ['a', 'b']

 ```

<h3>Contributors</h3>

This release contains contributions from (in alphabetical order):

Nathan Killoran, Robert Lang, Cedric Lin, Antal Száva

Reply:
Thanks Robert! I'll have a go including it